File: src/config/pilotConfig.js
/**
 * Jewellery Discovery & Intent Platform: Pilot Configuration
 */

module.exports = {
    // System Mode: PILOT or SCALE
    PILOT_MODE: process.env.PILOT_MODE === 'true' || true, // Default to true for MVP

    // Pilot Constraints
    PILOT_CONSTRAINTS: {
        MAX_SELLERS: 25,
        MAX_BUYERS: 1000,
        RESTRICTED_CITY: 'Mumbai',
        ALLOW_SELF_ONBOARDING: false
    },

    // Feature Flags - Admin Controlled
    FEATURES: {
        AI_TRY_ON: false,
        ONLINE_PURCHASE: false, // Core Hard Constraint: NO ECOMMERCE
        CUSTOM_FLOW: false,
        SELLER_INSIGHTS: false,
        BOOST_LISTINGS: false
    },

    /**
     * Helper to check if a feature is enabled
     */
    isFeatureEnabled: (featureName) => {
        return !!module.exports.FEATURES[featureName];
    }
};

File: src/config/pilot_config.py
import os

PILOT_MODE = os.environ.get('PILOT_MODE', 'true').lower() == 'true'

PILOT_CONSTRAINTS = {
    'MAX_SELLERS': 25,
    'MAX_BUYERS': 1000,
    'RESTRICTED_CITY': 'Chennai',
    'ALLOW_SELF_ONBOARDING': False
}

FEATURES = {
    'AI_TRY_ON': False,
    'ONLINE_PURCHASE': False, # Core Hard Constraint: NO ECOMMERCE
    'CUSTOM_FLOW': False,
    'SELLER_INSIGHTS': False,
    'BOOST_LISTINGS': False
}

def is_feature_enabled(feature_name: str) -> bool:
    return FEATURES.get(feature_name, False)
